### DISCUSSION TITLE
Heroku and upload of private files

### DISCUSSION THREAD

**Discussion by Maris1984** [strapi/strapi#8529] [Legacy Topics]
Could someone kindly advise on how to implement file storage for users' private files that should not be publicly available?
It's a Heroku hosted, Strapi + CRA system.
I expect to store up to 50Gb of users' private data, so storing in Mongo Atlas, which seems to be only alternative for now, would be a rather expensive solution.
Choosing an S3 bucket seems to be a cheaper alternative, however I cannot grant public access to files that end-users are going to upload.
I've read something about encryption, however, I'm unable to find any additional information on how to implement something like that. Perhaps there is some better alternatives to get this done?

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question, we are currently planning to decomission our GitHub Discussions in early 2021 and we don't want to lose the great questions asked by our community.

Questions should be directed to [our forum](https://forum.strapi.io/). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).
